engine/src/regex.cpp

```diff
@@ -356,15 +374,12 @@ regexp *MCR_compile(MCStringRef exp, bool casesensitive)
 	// If the pattern isn't found with the given flags, then create a new one.
 	if (re == nil)
 	{
-		/* UNCHECKED */ re = new(std::nothrow) regexp;
-		/* UNCHECKED */ re->pattern = MCValueRetain(exp);
-		re->flags = flags;
+		/* UNCHECKED */ re = new(std::nothrow) regex_t;
```

Comment:
@bwmilby from the looks of things you still need to set `re->re_pattern` and `re->re_flags` here. Also it would be good to sort out any unchecked return values while you are doing this cleanup.

Reply:
I moved those up to `regcomp`, partly to avoid having to release `pattern` on failure - I didn't save them until after the compile had returned a good result.

I'm thinking that the unchecked returns are pretty much going to succeed unless there is a memory/allocation issue.  In that case, I should be able to check for NULL and throw the memory error if the allocation wasn't successful.  I'll do some research and update.